compiler/codegen/OMRCodeGenPhase.cpp

```diff
@@ -419,39 +419,43 @@ void
 OMR::CodeGenPhase::performSetupForInstructionSelectionPhase(TR::CodeGenerator * cg, TR::CodeGenPhase * phase)
    {
    TR::Compilation *comp = cg->comp();
-   // TODO (GuardedStorage)
-#if defined(OMR_GC_CONCURRENT_SCAVENGER)
-   traceMsg(comp, "GuardedStorage: in performSetupForInstructionSelectionPhase\n");
 
-   auto mapAllocator = getTypedAllocator<std::pair<TR::TreeTop*, TR::TreeTop*> >(comp->allocator());
-
-   std::map<TR::TreeTop*, TR::TreeTop*, std::less<TR::TreeTop*>, TR::typed_allocator<std::pair<TR::TreeTop*, TR::TreeTop*>, TR::Allocator> >
-      currentTreeTopToappendTreeTop(std::less<TR::TreeTop*> (), mapAllocator);
-
-   for (TR::PreorderNodeIterator iter(comp->getStartTree(), comp); iter != NULL; ++iter)
+   if (cg->isConcurrentScavengeEnabled())
```

Comment:
I am also wondering about the value of this code in this (very) common phase in
the code generator.  As written, it would be very easy to sink all this logic into
its own codegen phase that would only be enabled on Z (should that be the right
thing to do).  Can you look into that?